cli/fmt.rs

```diff
@@ -234,21 +227,41 @@ fn is_supported(path: &Path) -> bool {
 
 pub fn collect_files(
   files: Vec<PathBuf>,
+  ignore: Vec<PathBuf>,
 ) -> Result<Vec<PathBuf>, std::io::Error> {
   let mut target_files: Vec<PathBuf> = vec![];
+  // FIXME the unwrap here panicks when we add a ignore file/folder which doesn't exist
+  let ignore: Vec<PathBuf> = ignore
+    .into_iter()
+    .map(|file| file.canonicalize().unwrap())
+    .collect();
 
   if files.is_empty() {
-    target_files.extend(files_in_subtree(
-      canonicalize_path(&std::env::current_dir()?)?,
-      is_supported,
-    ));
+    for entry in WalkDir::new(std::env::current_dir()?.canonicalize()?)
+      .into_iter()
+      .filter_entry(|e| {
+        !ignore.contains(&e.clone().path().canonicalize().unwrap())
```

Comment:
This isn't sufficient to match the current ignore semantics. Path prefixes are supported.